util/golomb.cpp

```diff
@@ -181,6 +181,11 @@ struct binary_output_file_iterator
         : file( file )
     {}
 
+    binary_output_file_iterator(const binary_output_file_iterator& other)                   = default; // Copy ctor
+    binary_output_file_iterator& operator= (const binary_output_file_iterator& other)       = default; // Copy assignment
+    binary_output_file_iterator(binary_output_file_iterator&& other)             noexcept   = default; // Move ctor
```

Comment:
Required to be movable, these four could be removed but it does make it more clear IMO, although they can be removed since the issue is resolved by making the member non-const

Reply:
The comments are not needed here. I think its quite obvious what is going on here.
Also I'm in favor to remove _not_ add defaults when they are not needed.

Reply:
Removed